More THM doco: last PPs and some components (#25023)
idaholab/moose: modules/thermal_hydraulics/doc/content/source/postprocessors/ADHeatStructureEnergy.md

# ADHeatStructureEnergy

!syntax description /Postprocessors/ADHeatStructureEnergy

The energy stored in these heat structures is calculated as:

!equation

> E = \int_\Omega \rho cp (T - T_{ref}) n_{ }units \text{plate depth} d\Omega

Author: ```suggestion
E = \int_\Omega \rho c_p (T - T_{ref}) n_{ }units \text{plate depth} d\Omega
```

> E = \int_\Omega \rho c_p (T - T_{ref}) n_{units} \text{plate depth} d\Omega

Reviewer: ```suggestion
E = \int_\Omega \rho c_p (T - T_{ref}) n_{units} d d\Omega
```

> where $E$ is the stored energy, $\rho$ the medium density, $cp$ the medium heat capacity,

Author: ```suggestion
where $E$ is the stored energy, $\rho$ the medium density, $c_p$ the medium specific heat capacity,
```

$T$ the medium temperature, $T_{ref}$ the reference temperature (reference for the enthalpy)

> and $n_{units}$ the number of units this heat structure represents and the plate depth is the
> size of uni-dimensional heat structures modeled.

Reviewer: ```suggestion
and $n_{units}$ the number of units this heat structure represents, and $d$ is the plate depth, i.e., the
size of uni-dimensional heat structures modeled.
```

> This volumetric integral is computed numerically using the system's quadrature.

Reviewer: Question because I don't know: are quadrature rules associated with systems? I remember something about block-dependent rules.

Author: so yeah no, not a system thing

> This volumetric integral is computed numerically using the system's quadrature.

Author: ```suggestion
This volumetric integral is computed numerically using the [local quadrature](syntax/Executioner/Quadrature/index.md)
```
